lib/internal/http2/core.js

```diff
@@ -1645,6 +1648,9 @@ class Http2Stream extends Duplex {
       return;
     }
 
+    if ((typeof data === 'string' || Buffer.isBuffer(data)) && !data.length)
```

Comment:
(Basically the same comment, except that the first piece of the conditional is *never* true; data is an array)